service/namespace_service.go

```diff
@@ -84,6 +85,34 @@ func (n *NamespaceService) ReconcileProjectNamespace(ctx context.Context, namesp
 				"object_kind": metricKindNamespace,
 			},
 		)
+	} else {
+		// check if the namespace has the correct labels
+		if !util.CompareLabels(nsResource.Labels, n.getResourceLabel(namespace, owner)) {
+			nsResource.Labels = n.getResourceLabel(namespace, owner)
+			err := util.UpdateResource(ctx, nsResource)
+			nsResource.Namespace = ControllerNamespace
```

Comment:
Why is this needed? A namespace object is not namespace scoped. The update will fail if you try to update the namespace field.

Reply:
util.UpdateResource has a custom logic to handle both namespace and cluster scoped objects. It would handle namespace update as well.